crates/shielder-scheduler-server/src/storage/dynamo_db.rs

```diff
@@ -0,0 +1,475 @@
+            .item(
+                "created_at",
+                AttributeValue::N(request.created_at.timestamp().to_string()),
+            )
+            .item(
+                "last_note_index",
+                AttributeValue::S(request.last_note_index.to_string()),
+            )
+            .item("payload", AttributeValue::S(payload));
+
+        if request.error_message.is_some() {
+            builder = builder.item(
+                "error_message",
+                AttributeValue::S(request.error_message.clone().unwrap()),
+            );
+        }
+        if let Some(processed_at) = request.processed_at {
+            builder = builder.item(
+                "processed_at",
+                AttributeValue::N(processed_at.timestamp().to_string()),
+            );
+        }
+
+        builder
+            .send()
+            .await
+            .map_err(|e| StorageError::Internal(format!("Failed to put item: {e}")))?;
+        Ok(())
+    }
+}
+
+impl StorageInterface for DynamoDb {
+    async fn insert_scheduled_request(
+        &self,
+        request: ScheduledRequest,
+    ) -> Result<(), StorageError> {
+        let payload = serde_json::to_string(&request)
+            .map_err(|e| StorageError::Internal(format!("Failed to serialize request: {e}")))?;
+
+        let mut builder = self
+            .client
+            .put_item()
+            .table_name(TABLE_NAME)
+            .condition_expression("attribute_not_exists(id)")
+            .item("id", AttributeValue::S(request.id.to_string()))
+            .item(
+                "status",
+                AttributeValue::S(status_to_str(&request.status).to_string()),
+            )
+            .item(
+                "relay_after",
+                AttributeValue::N(request.relay_after.timestamp().to_string()),
+            )
+            .item(
+                "created_at",
+                AttributeValue::N(request.created_at.timestamp().to_string()),
+            )
+            .item(
+                "last_note_index",
+                AttributeValue::S(request.last_note_index.to_string()),
+            )
+            .item("payload", AttributeValue::S(payload));
+
+        if let Some(err) = &request.error_message {
+            builder = builder.item("error_message", AttributeValue::S(err.clone()));
+        }
+        if let Some(processed_at) = request.processed_at {
+            builder = builder.item(
+                "processed_at",
+                AttributeValue::N(processed_at.timestamp().to_string()),
+            );
+        }
+
+        match builder.send().await {
+            Ok(_) => Ok(()),
+            Err(e) => {
+                if is_conditional_check_failed(&e) {
+                    Err(StorageError::DuplicateEntry(request.id))
+                } else {
+                    Err(map_internal("put_item", e))
+                }
+            }
+        }
+    }
+
+    async fn get_pending_requests(
+        &self,
+        limit: usize,
```

Comment:
I think interface `StorageInterface` lacks another method `get_all_pending_requests` that has not `limit`. It should up to caller of `StorageInterface` interface which method to choose. SO I'd propose to change `StorageInterface` 
```
async fn get_all_pending_requests(&self);
async fn get_some_pending_requests(&self, limit: usize);
```

Reply:
Actually, I think the API should look like
```
async fn get_pending_requests(&self, limit: Option<usize>);
```